core/src/main/java/io/questdb/std/IntLongPriorityQueue.java

```diff
@@ -90,25 +90,12 @@ public long popValue() {
         return v;
     }
 
-    private int binSearch(long v) {
-        if (size < 65) {
-            return scanSearch(v);
-        } else {
-            return binSearch0(v);
-        }
-    }
-
     @SuppressWarnings("StatementWithEmptyBody")
-    private int binSearch0(long v) {
+    private int binSearch(long v) {
         int low = 0;
         int high = size;
 
-        while (low < high) {
-
-            if (high - low < 65) {
-                return scanSearch(v);
-            }
-
+        while (high - low > 65) {
```

Comment:
Should be 64, not 65. Same with other modified classes.

Reply:
Should it depend on the target array entry size? I guess that for integer (4 byte) values number 64 were chosen with such logic in mind that linear scan will be performed over 3-4 cache lines (if it has size 128 bytes). 

If this assumption is right - then I guess that for double & long values (8 bytes) we need to halve region size lower bound...

Or this optimization has another reasoning?

Reply:
> Should it depend on the target array entry size? I guess that for integer (4 byte) values number 64 were chosen with such logic in mind that linear scan will be performed over 3-4 cache lines (if it has size 128 bytes).

I guess that was a best-effort attempt to optimize for small inputs. The size threshold is most likely empirically chosen on a concrete machine. I don't think 64 or 65 will make any difference, so let's keep it 65 considering that this number is already used in the native code.

Reply:
Yes, sure. 
Actually, from what I can understand, all fixed algorithms doesn't used in very critical high performance code (I guess thats why these bugs were unnoticed). So, the main motivation is to get rid of linear scan complexity just to avoid accidental increase of asymptotic due to future reuse of some of these classes.